src/main/scala/dsls/regex/Program.scala

```diff
@@ -152,10 +154,10 @@ object Program extends App {
    * the following several definitions with:
    *   val telNumber = '(' ~ digit{3} ~ ')' ~ digit{3} ~ '-' ~ digit{4}
    ***************************************************************************/
-  val threeDigits = Concat(digit, Concat(digit, digit))
-  val fourDigits = Concat(threeDigits, digit)
-  val areaCode = Concat(Literal('('), Concat(threeDigits, Literal(')')))
-  val telNumber = Concat(areaCode, Concat(threeDigits, Concat(Literal('-'), fourDigits)))
-  
+ val telNumber = '(' ~ digit{3} ~ ')' ~ digit{3} ~ '-' ~ digit{4}
+ val telNumber = digit{5} ~ '-' ~ digit{4}
+ 
   require(telNumber matches "(202)456-1111")
+  require(areaCode matches "95047-2827")        
```

Comment:
I think line 158 should be defining `areaCode`, but other than that this file looks good!